ush/python/pygfs/jedi/jedi.py

```diff
@@ -0,0 +1,274 @@
+#!/usr/bin/env python3
+
+import os
+from logging import getLogger
+from typing import List, Dict, Any, Optional
+from jcb import render
+from wxflow import (AttrDict,
+                    chdir, rm_p,
+                    parse_j2yaml,
+                    logit,
+                    Task,
+                    Executable,
+                    WorkflowException)
+
+logger = getLogger(__name__.split('.')[-1])
+
+
+class JEDI:
+
+    def __init__(self, task_config: AttrDict[str, Any], yaml_name: Optional[str] = None) -> None:
+
+        # For provenance, save incoming task_config as a private attribute of JEDI object
+        self._task_config = task_config
+
+        _exe_name = os.path.basename(task_config.JEDIEXE)
+
+        self.exe = os.path.join(task_config.DATA, _exe_name)
+        if yaml_name:
+            self.yaml = os.path.join(task_config.DATA, yaml_name + '.yaml')
+        else:
+            self.yaml = os.path.join(task_config.DATA, os.path.splitext(_exe_name)[0] + '.yaml')
+        self.config = AttrDict()
+        self.j2tmpl_dir = os.path.join(task_config.PARMgfs, 'gdas')
+
+    @logit(logger)
+    def set_config(self, task_config: AttrDict[str, Any], algorithm: Optional[str] = None) -> AttrDict:
+        """Compile a JEDI configuration dictionary from a template file and save to a YAML file
+
+        Parameters
+        ----------
+        task_config : AttrDict
+            Dictionary of all configuration variables associated with a GDAS task.
+        algorithm (optional) : str
+            Name of the algorithm used to generate the JEDI configuration dictionary.
+            It will override the algorithm set in the task_config.JCB_<>_YAML file.
+
+        Returns
+        ----------
+        None
+        """
+
+        if 'JCB_BASE_YAML' in task_config.keys():
+            # Step 1: Fill templates of the JCB base YAML file
+            jcb_config = parse_j2yaml(task_config.JCB_BASE_YAML, task_config)
+
+            # Step 2: If algorithm is present then override the algorithm in the JEDI
+            #         config. Otherwise, if the algorithm J2-YAML is present, fill
+            #         its templates and merge.
+            if algorithm:
+                jcb_config['algorithm'] = algorithm
+            elif 'JCB_ALGO' in task_config.keys():
+                jcb_config['algorithm'] = task_config.JCB_ALGO
+            elif 'JCB_ALGO_YAML' in task_config.keys():
+                jcb_algo_config = parse_j2yaml(task_config.JCB_ALGO_YAML, task_config)
+                jcb_config = {**jcb_config, **jcb_algo_config}
```

Comment:
Cleaner if you use the `dict.update()` method rather than unpacking everything:
```suggestion
                jcb_config.update(jcb_algo_config)
```